docs/intro.md

```diff
@@ -0,0 +1,72 @@
+
+One of the larger challenges is the pervasive use of `System.Web.HttpContext` throughout a code base to access information about a request or update a response. Until now, in order to move forward, a large scale rewrite was required to remove this dependency. Now, the adapters in this repo provide a set of runtime helpers to access the types your code is expecting but in a way that will work on ASP.NET Core with little change.
+
+A complete migration may take a while (sometimes a multi-year effort) depending on the size of the application. In order to continue deploying an application to production while working on migrating, the best pattern is to follow is the [Strangler Fig pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/strangler-fig). This pattern allows for continual development on the old system with an incremental approach to moving forward. This document will describe how to apply that pattern to an ASP.NET app migrating towards ASP.NET Core.
+
+## Migration Journey
+
+When starting a migration journey, the application will be targeting ASP.NET Framework and running on Windows with its supporting libraries:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> framework[Framework App]
+  framework --- libraries[[Business logic]]
+```
+
+The first step is to introduce a new application based on ASP.NET Core that will become the entry point. Traffic will enter the core app and if the core app cannot match a route, it will proxy the request onto the framework app via [YARP](https://microsoft.github.io/reverse-proxy/) and serve a response as the application has already been doing. Majority of code will continue to be in the framework app, but the core app is now set up to start migrating routes to:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> core[ASP.NET Core App]
+  core --- libraries
+  core -- YARP proxy --> framework[ \n\nFramework App\n\n\n]
+  framework --- libraries[[Business logic]]
+```
+
+In order to start moving over business logic that relies on `HttpContext`, the libraries need to be built against `Microsoft.AspNetCore.SystemWebAdapters`. This will ensure that the libraries are using surface area that is available on both ASP.NET and ASP.NET Core:
+
+```mermaid
+flowchart LR;
+  external[External Traffic] --> core[ASP.NET Core App]
+  core -- Adapters --- libraries
+  core -- YARP proxy --> framework[ \n\nFramework App\n\n\n]
+  framework --- libraries[[Business logic]]
+```
+
+At this point, the journey is to focus on moving routes over one at a time. This could be WebAPI or MVC controllers (or even a single method from a controller), ASPX pages, handlers, or some other implementation of a route. If the route is available in the core app, it will then be matched and served from there. Over time, the core app will start processing more of the routes served than the framework app:
```

Comment:
At this time, you theoretically have the route in both ASP.NET Core and ASP.NET Framework apps.

I think it would be nice to highlight the advantages of being able to easily test code side-by-side in production (before cutting over). and to be able to easily rollback a release by changing the registered routes.